winpr/libwinpr/utils/wlog/wlog.c

```diff
@@ -380,34 +381,39 @@ int WLog_ParseFilters()
 	if (!env)
 		return -1;
 
-	nSize = GetEnvironmentVariableA("WLOG_FILTER", env, nSize);
-	count = 1;
-	p = env;
-
-	while ((p = strchr(p, ',')) != NULL)
 	{
-		count++;
-		p++;
-	}
+		nSize = GetEnvironmentVariableA("WLOG_FILTER", env, nSize);
+		count = 1;
+		p = env;
 
-	g_FilterCount = count;
-	p = env;
-	count = 0;
-	strs = (char**) calloc(g_FilterCount, sizeof(char*));
+		while ((p = strchr(p, ',')) != NULL)
+		{
+			count++;
+			p++;
+		}
 
-	if (!strs)
-	{
-		free(env);
-		return -1;
+		g_FilterCount = count;
+		p = env;
 	}
+	{
+		count = 0;
+		strs = (LPCSTR*) calloc(g_FilterCount, sizeof(LPCSTR));
 
-	strs[count++] = p;
+		if (!strs)
+		{
+			free(env);
+			return -1;
+		}
 
-	while ((p = strchr(p, ',')) != NULL)
-	{
-		strs[count++] = p + 1;
-		*p = '\0';
-		p++;
+		strs[count++] = p;
+
+		while ((p = strchr(p, ',')) != NULL)
+		{
+			if ((p + 1) && (count < g_FilterCount))
```

Comment:
again: p + 1 is always true, probably that should be if (p[1] ...)  or  if (*(p+1) ...)